hphp/hack/src/errors/typing_error.ml

```diff
@@ -4342,14 +4343,26 @@ module Primary = struct
       lazy [(pos2, "Mixed with this kind of field (key => value)")],
       [] )
 
-  let unbound_name_typing pos name =
+  let unbound_name_typing pos name class_exists =
+    let quickfixes =
+      match class_exists with
+      | true ->
+        let subst = String.sub name ~pos:1 ~len:(String.length name - 1) in
+        [
+          Quickfix.make
+            ~title:("Add " ^ Markdown_lite.md_codify "new")
+            ~new_text:("new " ^ subst)
```

Comment:
Rather than re-inserting the name of the class, I suggest you just insert the `"new "`. You can create an empty position where the start and end are the same, so you don't overwrite the existing name.